codegen/src/main/java/software/amazon/awssdk/codegen/CodeGenerator.java

```diff
@@ -66,12 +57,14 @@ public static Builder builder() {
      * code.
      */
     public void execute() {
+        // Make sure ClassName is statically initialized before we do anything in parallel.
+        // Parallel static initialization of ClassName and TypeName can result in a deadlock:
+        // https://github.com/square/javapoet/issues/799
+        ClassName.get(Object.class);
+
```

Comment:
Should this be moved to a `static` block within this class?

Reply:
Sure

Reply:
Fixed.